control-plane-operator/controllers/hostedcontrolplane/olm/assets.go

```diff
@@ -68,8 +67,8 @@ func MustDeployment(fileName string) *appsv1.Deployment {
 	return deployment
 }
 
-func MustCronJob(fileName string) *batchv1beta1.CronJob {
-	cronJob := &batchv1beta1.CronJob{}
+func MustCronJob(fileName string) *batchv1.CronJob {
+	cronJob := &batchv1.CronJob{}
```

Comment:
This appears to silently fail and then cause a NPD down the line when accessing the image:
```
{"level":"info","ts":"2022-05-06T12:11:52Z","logger":"controller.hostedcontrolplane","msg":"Reconciling OLM","reconciler group":"hypershift.openshift.io","reconciler kind":"HostedControlPlane","name":"example-nxnnk","namespace":"e2e-clusters-8c2cb-example-nxnnk"}
panic: runtime error: index out of range [0] with length 0

goroutine 1443 [running]:
github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/olm.reconcileCatalogCronJob(0xc000962600, {0xc00071f130}, {0xc000be7a00, 0x7f}, 0xc000962000)
	/hypershift/control-plane-operator/controllers/hostedcontrolplane/olm/catalogs.go:115 +0x1ba
github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/olm.ReconcileCertifiedOperatorsCronJob(0xc000962600, {0xc00071f130}, {0xc000be7a00, 0x7f})
	/hypershift/control-plane-operator/controllers/hostedcontrolplane/olm/catalogs.go:100 +0x4e
github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane.(*HostedControlPlaneReconciler).reconcileOperatorLifecycleManager.func12()
	/hypershift/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:2054 +0x6a
github.com/openshift/hypershift/support/upsert.mutate(0xc000cb4db0, {{0xc0006a69a0, 0x20}, {0x333665f, 0x23}}, {0x373fec0, 0xc000962600})
	/hypershift/support/upsert/upsert.go:132 +0x71
github.com/openshift/hypershift/support/upsert.(*createOrUpdateProvider).CreateOrUpdate(0xc000010128, {0x372d568, 0xc00057d770}, {0x37345b8, 0xc000220aa0}, {0x373fec0, 0xc000962600}, 0xc000cb4db0)
	/hypershift/support/upsert/upsert.go:67 +0x17b
github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane.(*HostedControlPlaneReconciler).reconcileOperatorLifecycleManager(0xc000220aa0, {0x372d568, 0xc00057d770}, 0xc0006f2a80, 0xc000a08240, {0xc000715d20, 0xe})
	/hypershift/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:2053 +0x1eb0
github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane.(*HostedControlPlaneReconciler).update(0xc000220aa0, {0x372d568, 0xc00057d770}, 0xc0006f2a80)
	/hypershift/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:687 +0x35d3
github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane.(*HostedControlPlaneReconciler).Reconcile(0xc000220aa0, {0x372d568, 0xc00057d770}, {{{0xc000a24c80, 0x20}, {0xc000a4b6e0, 0xd}}})
	/hypershift/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:491 +0x436c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc0001964d0, {0x372d568, 0xc00057d770}, {{{0xc000a24c80, 0x20}, {0xc000a4b6e0, 0xd}}})
	/hypershift/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:114 +0x365
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001964d0, {0x372d568, 0xc00057c9f0}, {0x2fefae0, 0xc0003ac1c0})
	/hypershift/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:311 +0x425
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001964d0, {0x372d4c0, 0xc000138e40})
	/hypershift/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266 +0x398
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/hypershift/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227 +0xdc
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/hypershift/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:223 +0x4d6
```

Reply:
weird!!!!!

Reply:
oh I know!

Reply:
Needed to update the yamls that are defined for the cronjobs

Reply:
fixed!

Reply:
I am pretty sure it is more than just changing the apiVersion field. The golang unmarshaler silently drops fields that are not present in the target go struct, so I would assume that is what happened after the schema changed between v1beta2 and v1.

Reply:
oh really?

Reply:
I will do a full build in our environment and make sure the cronjob looks good although in our environment only the olm-collector job runs

Reply:
there's no rush on this pr I just wanted to clean it up because I made the original change back in the day to move it to being old versions